### ISSUE TITLE
Security officer crew members won't use a ranged weapon if you attack them in multiplayer

### ISSUE THREAD

**Issue by Rokvach** [Regalis11/Barotrauma#8065]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
If you attack a security officer bot the bot wont attack back if they try to use a ranged weapon to do so

**Steps To Reproduce**
1. Start a multiplayer round
2. Spawn in a security officer
3. Take control of security officer and remove all melee weapons from their inventory (I also removed handcuffs)
4. Control your main character again and shoot at the security officer
5. Notice that the security officer will just stand there pointing their weapon at you

Also it seems like even if the security officer dies and you use the revive command on them they will still be aggroed on you and won't do anything else. 

**Version**
v0.16.0.0
Branch: dev
Commit: https://github.com/Regalis11/Barotrauma-development/commit/feaadf91032080d341c1e9a327306ccf20120a9e

**Additional information**
![image.png](https://images.zenhubusercontent.com/5c924117534a9b0a86cdddce/6bd855ba-7eda-4518-b562-87a65e7c773e)

**Comment by Regalis11** [comment]
Probably caused by the changes in https://github.com/Regalis11/Barotrauma/issues/7956

**Comment by itchyOwl** [comment]
It does shoot, but the threshold is quite high. So the question is: is it too high? The security doesn't seem to warn you, which is a bug.

**Comment by itchyOwl** [comment]
Seems that there's something wrong there: they don't stop pointing with the gun, unless you'll keep attacking.

**Comment by itchyOwl** [comment]
Fixed in Regalis11/Barotrauma-development@fef2880 and Regalis11/Barotrauma-development@7eaf473. Also refactored and fixed some related issues. Requires extensive testing, both single player and multiplayer. Also in outposts.

**Comment by Rokvach** [comment]
I was still able to reproduce the issue, security guards onboard your sub will attack you with melee weapons but not with ranged weapons no matter how much damage you cause to them.

**Comment by itchyOwl** [comment]
Hmm, I can't repro: in multiplayer the security first tries to arrest and then attack, if I do further damage (If they don't have stunners, they just point at you for a while when they are in the "arrest" state). Also tested it by taking all the melee weapons from the guards before attacking them.

What do you mean it works correctly in single player? Do they attack you or not? Because in single player the behavior is now different from what it used to be and what is in the multiplayer: they shouldn't ever attack you or arrest you.

**Comment by Rokvach** [comment]
In singleplayer they don't attack which is correct, in multiplayer they do attack with melee weapons and handcuffs you but if you take those items out they just aim ranged weapons at you and doesn't do anything no matter how much damage I do to them.

**Comment by itchyOwl** [comment]
Ok, but I still wasn't able to repro it.

**Comment by Rokvach** [comment]
Looks like they won't shoot at you if you have godmode on, I did all my tests in godmode and now when I turned it off they do shoot but takes a very long time to do so. 

Also it feels like there's something broken in bandit weapon usage as well, they don't immediately start shooting at you but instead waits for a long time and then fires for a short time. They also run away which I don't remember happening before.
